### ISSUE TITLE
Cannot add tracks to owned playlist when logging in via email

### ISSUE THREAD

**Issue by xou816** [xou816/spot#234]
**Describe the bug**
Cannot add tracks to owned playlist when logging in via email. (thanks NoTuxNoBux for bringing that up! https://github.com/xou816/spot/issues/186#issuecomment-862997115 )

**To Reproduce**
Create a new playlist in your account (using another client than spot)
Log in to Spot using your email, not your username
Select songs and hit "+".

**Expected behavior**
the contextual menu only allows adding songs to the play queue when it should be possible to add them to the newly created playlist as well.

**General information:**
 - Distribution: any
 - Installation method [e. g. built from source, installed from Flathub...]: any
 - Version [e.g. 0.1.0]: 0.1.14 and above
 - Device used [e. g. desktop, phone...]: any

**Comment by xou816** [comment]
The Credentials object should not be populated with the "username" used to login (which could be an email) as seen here https://github.com/xou816/spot/blob/master/src/player/player.rs#L133 but with `session.username()` instead

**Comment by NoTuxNoBux** [comment]
As mentioned in #186, this still seems to be a problem for me.

About my account: it might be special in the sense that it is an account that has been attached to someone else's family subscription, i.e. I was invited and created my own account that immediately became part of this subscription. I can log in with both my own username and email address both in the web version of Spotify and Spot itself.

**Comment by xou816** [comment]
Thanks for helping with this :)

I am out of ideas for now, can we check again what we did check last time? https://github.com/xou816/spot/issues/186#issuecomment-862427026 just to make sure that we were talking about the same thing :)

So for instance one my playlists' JSON looks like that (simplified):

```json
{
  "id" : "",
  "images" : [],
  "name" : "",
  "owner" : {
    "display_name" : "xou816",
    "external_urls" : {
      "spotify" : "https://open.spotify.com/user/xou816"
    },
    "href" : "https://api.spotify.com/v1/users/xou816",
    "id" : "xou816",
    "type" : "user",
    "uri" : "spotify:user:xou816"
  },
  "tracks" : {}
}
```

... where `display_name` matches exactly my username. If its doesn't for you, maybe another field does? (id?)

Then we can compare it with the username associated with your session, which you see using Seahorse for instance:

![image](https://user-images.githubusercontent.com/6277494/124314610-31941b80-db62-11eb-902f-70224d1b560f.png)

**Comment by NoTuxNoBux** [comment]
Thanks for taking another look at this.

You are right, it still doesn't match my username, but it does match my display name - the name I chose for my account. The username I have listed in Spotify is some sort of hash that it seemingly automatically generated, possibly because I joined a family account. I can log in with that hash, but it's different from my chosen name, and cannot log in with the latter.

So to post some examples with some fake names, it currently looks like this:

* Username in Spotify: `dsnd89052d8902436842096a26890`
* Email address in Spotify: `notuxnobux@example.com`
* Display name in Spotify: `NoTuxNoBux`
* `owner.display_name` in Spot: `NoTuxNoBux`
* In Spot, I can log in with: `dsnd89052d8902436842096a26890` or `notuxnobux@example.com`

In other words: you are right, the match is still not happening. Apologies, I should have noticed this :slightly_smiling_face: .

**Comment by xou816** [comment]
No worries, and again thanks for taking the time to help me with this!

Also, that's really on me for using something that's called display_name for more than it is...

On the playlist side, does the owner.id or the owner.uri field match (partially in the latter case) with your username (dsnd...)?

**Comment by NoTuxNoBux** [comment]
No problem! And yes, the `owner.id` seems to match the hash generated by Spotify that is my username.

The owner.uri field is formatted as `spotify:user:HASH`, so i.e. `spotify:user:dsnd89052d8902436842096a26890`.

**Comment by xou816** [comment]
Interesting, so maybe this might be the actual fix!

Would you mind trying the artifact/build from this page if you have some time? https://github.com/xou816/spot/actions/runs/1001830044

**Comment by NoTuxNoBux** [comment]
Yes, I can confirm this fixes the issue! I can now add tracks to my playlists, as well as remove items from them. Thanks!

**Comment by xou816** [comment]
Perfect, then I am merging the fix! Thanks a lot! :)
